### PR TITLE
[RDY] Better docs for translators

### LINKED ISSUE #21308: Write better documentation for translators
Since there's a web translation system in place, `CONTRIBUTING.md` should be updated with links to the translation project and to the `TRANSLATING.md` document. Also, the latter should contain a guide for translators and, in general, be updated with current information.

Original text:
> Hello, I've noticed that most (if not all) of the translation work is done by a couple of people, and, not to diminish their work, but I think using a web translation tool would attract more contributors to the translation effort, or at least make the current translators' work easier.
>
> These sort of tools would allow people who are not comfortable using `git` and `gettext` directly to suggest and review translations using a web interface, without having to partake into the black magic of the translation scripts.
>
> I know two platforms which provide free hosting for open source projects:
>
> * [Weblate](https://weblate.org/)
> * [Transifex](https://www.transifex.com/)
>
> Out of these, I would prefer Weblate, since their offer for open source projects is more explicit, and I know Minetest, other free game using `gettext`, is using it succesfully.

### PR DESCRIPTION
**Pull request by fernzi**
I've wrote better, up-to-date documentation for translators.

### Changes

* Added links to the translation platform in `CONTRIBUTING.md`.
* Removed outdated information from `TRANSLATING.md`.
* Created a small visual guide on how to use Transifex.
* Moved the stuff for developers from `CONTRIBUTING.md` into a developers' section in `TRANSLATING.md`.

### Closed issues

* Closes #21308.

### CODE REVIEW

**Comment by BrettDong** [comment]
The language list is incomplete, eg. Polish isn't present.

File: doc/TRANSLATING.md [hunk 1]
```diff
@@ -1,47 +1,42 @@
...
 
-You can lend your support to a language translation project:
+Some of the currently supported languages are:
 
-On [Transifex](https://www.transifex.com/projects/p/cataclysm-dda/):
-* Argentinian
-* Chinese (Simplified)
-* Chinese (Traditional)
+* Chinese (China)
+* Chinese (Taiwan)
```
Comment (BrettDong) [inline doc/TRANSLATING.md]: Those two should be 'Simplified Chinese' and 'Traditional Chinese'.

Comment (franzzapata) [inline doc/TRANSLATING.md]: This is how they are listed on Transifex, should I still change them to Simplified and Traditional?

Comment (BrettDong) [inline doc/TRANSLATING.md]: Yes. 'Traditional' and 'Simplified' names avoids political disputes.

**Comment by franzzapata** [comment]
I've updated the supported languages list.

Now, if there's no objection, I'm going to clean up and move the stuff about using `gettext` from `CONTRIBUTING.md` into a "Developers" section in `TRANSLATING.md`, since IMO, it doesn't belong in the contributing file, and having everything pertaining translation in one place would be very handy.

**Comment by franzzapata** [comment]
I'd say it's done, unless someone has any other suggestion.

**Comment by Coolthulhu** [comment]
Needs merge conflict resolution

**Comment by franzzapata** [comment]
@Coolthulhu Done!

**Comment by AMurkin** [comment]
Please consider the possibility of bringing images to a more mobile-prepared form - less aspect ratio, larger meaningful content.

**Comment by Leland** [comment]
As a counterpoint, CDDA is chiefly a desktop application and this is documentation for development, which also chiefly (if not only) takes places on the desktop.

**Pull request merged**